internal/commands/user/flags.go

```diff
@@ -35,5 +51,101 @@ var (
 		providerTypeGoogle,
 		providerTypeAnonymous,
 		providerTypeCustom,
+		providerTypeApple,
+		providerTypeCustomFunction,
+	}
+)
+
+// statusType enumset
```

Comment:
two contradicting things here:

where did we land on bringing this stuff into the `realm` package?  there was a comment resolved about this, but no response and seemingly no movement here?

if we're keeping this stuff here, then I think we can remove a lot of these docstrings (at least on the unexported types and their functions, the most uninteresting ones)